### PR TITLE
JC-1269 Allow deleting own topic if others didn't answer

### PR DESCRIPTION
**Pull request by artexnet**
- User can delete own starting post if no other user has added a reply, or those replies were removed/deleted by their owners
- "Delete" button is also visible when there are several posts added only by initial topic starter

### CODE REVIEW

**Comment by ctapobep** [comment]
@mixas27 would you please do review one more time?

File: jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java [hunk 1]
```diff
@@ -785,4 +785,18 @@ public Post getDraftForUser(JCUser user) {
     public int getDisplayedPostsCount() {
         return getDisplayedPosts().size();
     }
+
+	/**
+	 * Checks if this topic contains only posts added by topic Owner.
+	 *
+	 * @return <code>true</code> if condition is followed, otherwise <code>false</code>
+	 */
+	public boolean isContainsOwnerPostsOnly() {
```
Comment (mixas27) [inline jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java]: Don't forget about unit tests

**Comment by artexnet** [comment]
Thanks @mixas27 , unit tests are added.

**Comment by mixas27** [comment]
Please squash commits into single one and then I' merge

**Comment by artexnet** [comment]
@mixas27 done.

**Comment by mixas27** [comment]
Merged. Thank you.

**Pull request closed without merging**